geoh5py/objects/drillhole.py

```diff
@@ -219,19 +213,16 @@ def surveys(self):
         :obj:`numpy.array` of :obj:`float`, shape (3, ): Coordinates of the surveys
         """
         if (getattr(self, "_surveys", None) is None) and self.existing_h5_entity:
-            surveys = self.workspace.fetch_coordinates(self.uid, "surveys")
-            if len(surveys) == 1:
-                self.surveys = np.vstack(
-                    [
-                        np.c_[0, surveys["Azimuth"], surveys["Dip"]],
-                        surveys.view("<f4").reshape((-1, 3)),
-                    ]
-                )
-            else:
-                self._surveys = surveys
+            self._surveys = self.workspace.fetch_coordinates(self.uid, "surveys")
 
         if getattr(self, "_surveys", None) is not None:
-            return self._surveys
+            surveys = self._surveys.view("<f4").reshape((-1, 3))
+
+            # Repeat first survey point at surface for de-survey interpolation
+            surveys = np.vstack([surveys[0, :], surveys])
+            surveys[0, 0] = 0.0
+
+            return surveys.astype(float)
```

Comment:
**question:** is this always adding an extra point, or only if needed?

Reply:
Always yes, just because it doesn hurt, but will always fix the case where the survey point(s) is only defined below the collar. Has to do with the searchsorted needing a point on the "left"